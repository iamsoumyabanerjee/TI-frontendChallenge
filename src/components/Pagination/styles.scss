@import '../../styles/variables.scss';
@import '../../styles/breakpoint.scss';

.pagination {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  @include viewport(medium) {
    justify-content: left;
  }
  .current-page {
    background-color: $white;
    display: block;
    width: 38px;
    height: 38px;
    line-height: 38px;
    text-align: center;
  }
}