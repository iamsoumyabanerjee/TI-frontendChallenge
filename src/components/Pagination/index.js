import React from 'react'
import { SEARCH_TYPE_PAGE } from '../../utils/constants'

import './styles.scss'
const Pagination = ({ currentPage, pageCount, onClick }) => {
  const NEXT = 'NEXT'
  const PREV = 'PREV'

  const onClickHandler = type => {
    let updatedPage = parseInt(currentPage.value)
    if (type == NEXT) {
      updatedPage = updatedPage + 1
    } else {
      updatedPage = updatedPage - 1
    }
    onClick({ value: updatedPage }, SEARCH_TYPE_PAGE)
  }

  return (
    <div className='pagination'>
      <button disabled={currentPage.value < 2} onClick={() => onClickHandler(PREV)}>Prev</button>
      <span className='current-page'>{currentPage.value}</span>
      <button disabled={currentPage.value == pageCount} onClick={() => onClickHandler(NEXT)}>Next</button>
    </div>
  )
}

export default Pagination