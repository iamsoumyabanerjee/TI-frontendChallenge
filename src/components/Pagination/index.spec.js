import React from 'react'
import renderer from 'react-test-renderer'
import Pagination from '.'

describe('#Error', () => {
  it('should render correctly', () => {
    const props = {
      currentPage: { value: 1 }, pageCount: 10, onClick: jest.fn()
    }
    const tree = renderer
      .create(<Pagination { ...props } />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})