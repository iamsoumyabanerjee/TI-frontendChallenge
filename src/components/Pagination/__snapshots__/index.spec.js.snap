// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Error should render correctly 1`] = `
<div
  className="pagination"
>
  <button
    disabled={true}
    onClick={[Function]}
  >
    Prev
  </button>
  <span
    className="current-page"
  >
    1
  </span>
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
